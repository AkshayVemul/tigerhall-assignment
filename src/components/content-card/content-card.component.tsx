import { Box, VStack } from '@chakra-ui/react'

import {
  ContentCardDescription,
  ContentCardDescriptionProps,
} from './components/content-card-description'
import {
  ContentCardAuthor,
  ContentCardAuthorProps,
} from './components/content-card-author'
import { ContentCardFooter } from './components/content-card-footer'
import {
  ContentCardBody,
  ContentCardBodyProps,
} from './components/content-card-body'

export type ContentCardProps = ContentCardBodyProps &
  ContentCardDescriptionProps &
  ContentCardAuthorProps

export const ContentCard = ({
  title,
  description,
  authorName,
  authorCompany,
  imageUrl,
  completed,
  watchedMinutes,
}: ContentCardProps) => {
  return (
    <Box bg={'white'} width={'244px'} borderRadius={'8'} overflow={'hidden'}>
      <ContentCardBody
        imageUrl={imageUrl}
        completed={completed}
        watchedMinutes={watchedMinutes}
      />

      <VStack p={2} pb={3} gap={2} alignItems={'flex-start'}>
        <ContentCardDescription title={title} description={description} />
        <ContentCardAuthor
          authorName={authorName}
          authorCompany={authorCompany}
        />
        <ContentCardFooter />
      </VStack>
    </Box>
  )
}
