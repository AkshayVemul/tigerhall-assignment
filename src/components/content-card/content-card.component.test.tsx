import { render } from '@testing-library/react'
import { ContentCard } from './content-card.component'

describe('Content Card', () => {
  it('should render the component', () => {
    const { container } = render(
      <ContentCard
        authorName="Author Name"
        authorCompany="Author Company"
        title="Title"
        description="Lorem ipsum dolor, sit"
        imageUrl="https://www.test_url.com"
        completed={30}
        watchedMinutes={20}
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
