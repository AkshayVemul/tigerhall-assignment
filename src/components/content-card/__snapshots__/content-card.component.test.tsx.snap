// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Card should render the component 1`] = `
<div>
  <div
    class="css-20pw3a"
  >
    <div
      class="css-79elbk"
    >
      <img
        alt="podcast_image"
        class="chakra-image css-ikrqgh"
        loading="lazy"
        src="https://www.test_url.com/resize/250x//"
      />
      <div
        class="chakra-stack css-edma2d"
      >
        <svg
          class="chakra-icon css-9mzut3"
          focusable="false"
          viewBox="0 0 12 12"
        >
          <path
            clip-rule="evenodd"
            d="M6 0C4.81331 0 3.65328 0.351894 2.66658 1.01118C1.67989 1.67047 0.910851 2.60754 0.456725 3.7039C0.0025997 4.80026 -0.11622 6.00665 0.115291 7.17054C0.346802 8.33443 0.918247 9.40352 1.75736 10.2426C2.59648 11.0818 3.66557 11.6532 4.82946 11.8847C5.99335 12.1162 7.19975 11.9974 8.2961 11.5433C9.39246 11.0891 10.3295 10.3201 10.9888 9.33342C11.6481 8.34672 12 7.18669 12 6C11.9982 4.40926 11.3655 2.88419 10.2406 1.75936C9.11581 0.634539 7.59074 0.00181488 6 0ZM6 11.1429C4.63603 11.1429 3.32793 10.601 2.36345 9.63655C1.39898 8.67207 0.857145 7.36397 0.857145 6C0.857145 4.63603 1.39898 3.32792 2.36345 2.36345C3.32793 1.39898 4.63603 0.857143 6 0.857143V6L9.63472 9.63471C9.15805 10.1131 8.59157 10.4926 7.96781 10.7514C7.34406 11.0102 6.67532 11.1432 6 11.1429Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="chakra-text css-1y937cb"
        >
          30
          % Completed
        </p>
      </div>
      <div
        class="chakra-stack css-15jjwtq"
      >
        <svg
          class="chakra-icon css-14ixh0p"
          focusable="false"
          viewBox="0 0 14 14"
        >
          <path
            d="M14 6.92699C14 3.34967 11.1815 0.316553 7.58366 0.0225113C5.67447 -0.133324 3.74932 0.525302 2.30492 1.83106C0.766869 3.2213 -0.0712142 5.17461 0.00475042 7.18986C0.0276278 7.79845 0.13464 8.39433 0.312883 8.96806C0.308567 9.00866 0.299936 9.04803 0.299936 9.08986V11.603C0.299936 12.2796 0.882527 12.8333 1.59459 12.8333H3.66561C3.81019 12.8333 3.92885 12.7205 3.92885 12.5831V8.1097C3.92885 7.97231 3.81018 7.85953 3.66561 7.85953H1.59459C1.51994 7.85953 1.44742 7.86897 1.37622 7.88086C1.33566 7.63848 1.3076 7.39365 1.29854 7.14555C1.23553 5.48218 1.92774 3.87006 3.19693 2.7226C4.37334 1.66002 5.93255 1.12238 7.47233 1.24827C10.4069 1.48859 12.7053 3.98283 12.7053 6.92697C12.7053 7.24971 12.6773 7.56836 12.6246 7.88086C12.553 7.86897 12.4805 7.85953 12.4054 7.85953H10.3344C10.1894 7.85953 10.0711 7.97231 10.0711 8.1097V12.5831C10.0711 12.7205 10.1894 12.8333 10.3344 12.8333H12.4054C13.1175 12.8333 13.7001 12.2796 13.7001 11.603V9.08986C13.7001 9.04967 13.6919 9.01194 13.6876 8.97257C13.8921 8.31681 14 7.6303 14 6.92698L14 6.92699Z"
            fill="currentColor"
            xmlns="http://www.w3.org/2000/svg"
          />
        </svg>
      </div>
      <div
        class="chakra-stack css-makbeh"
      >
        <svg
          class="chakra-icon css-1xfda0m"
          focusable="false"
          viewBox="0 0 12 12"
        >
          <path
            clip-rule="evenodd"
            d="M3.66519 11.5287C2.95042 11.2263 2.3083 10.7943 1.75749 10.2425C1.20668 9.6917 0.773687 9.05056 0.471281 8.33579C0.159057 7.59647 0 6.811 0 6.00098C0 5.19097 0.159057 4.40452 0.471281 3.66519C0.773687 2.95042 1.20668 2.3083 1.75749 1.75749C2.3083 1.20668 2.95042 0.773687 3.66519 0.471281C4.40353 0.158076 5.18999 0 6.00098 0C6.811 0 7.59647 0.158076 8.33579 0.471281C9.05056 0.773687 9.6917 1.20668 10.2425 1.75749C10.7943 2.3083 11.2263 2.95042 11.5287 3.66519C11.8419 4.40452 12 5.19097 12 6.00098C12 6.811 11.8419 7.59647 11.5287 8.33579C11.2263 9.05056 10.7943 9.6917 10.2425 10.2425C9.6917 10.7943 9.05056 11.2263 8.33579 11.5287C7.59647 11.8419 6.811 12 6.00098 12C5.18999 12 4.40452 11.8419 3.66519 11.5287ZM4.26482 1.89387C3.73365 2.11773 3.25746 2.43978 2.84803 2.8492C2.4386 3.25764 2.11754 3.73482 1.8927 4.26599C1.66001 4.81484 1.54219 5.39903 1.54219 6.00089C1.54219 6.60276 1.66001 7.18597 1.8927 7.7358C2.11754 8.26599 2.4386 8.74316 2.84803 9.15161C3.25746 9.56201 3.73365 9.88307 4.26482 10.1079C4.81465 10.3396 5.39786 10.4574 6.00071 10.4574C6.60257 10.4574 7.18578 10.3396 7.73561 10.1079C8.2658 9.88307 8.74297 9.56201 9.1524 9.15161C9.56084 8.74316 9.88288 8.26599 10.1067 7.7358C10.3394 7.18597 10.4573 6.60276 10.4573 6.00089C10.4573 5.39903 10.3394 4.81484 10.1067 4.26599C9.88288 3.73482 9.56084 3.25764 9.1524 2.8492C8.74297 2.43978 8.2658 2.11773 7.73561 1.89387C7.18578 1.66118 6.60257 1.54336 6.00071 1.54336C5.39786 1.54336 4.81465 1.66118 4.26482 1.89387ZM5.9145 7.1246C5.48838 7.1246 5.14277 6.77899 5.14277 6.35287V3.17172C5.14277 2.74561 5.48838 2.4 5.9145 2.4C6.34061 2.4 6.68622 2.74561 6.68622 3.17172V5.58115H8.14326C8.56938 5.58115 8.91499 5.92675 8.91499 6.35287C8.91499 6.77899 8.56938 7.1246 8.14326 7.1246H5.9145Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="chakra-text css-1jueml0"
        >
          20
          m
        </p>
      </div>
      <div
        class="css-11deuae"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="30"
          class="css-1jy0d03"
          role="progressbar"
          style="width: 30%;"
        />
      </div>
    </div>
    <div
      class="chakra-stack css-5czzcw"
    >
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-hdt7pk"
        >
          Title
        </p>
        <p
          class="chakra-text css-12wraxb"
        >
          Lorem ipsum dolor, sit
        </p>
      </div>
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-1wghd8i"
        >
          Author Name
        </p>
        <p
          class="chakra-text css-1eqkd9p"
        >
          Author Company
        </p>
      </div>
      <div
        class="chakra-stack css-1ort7sh"
      >
        <a
          href="#"
        >
          <svg
            class="chakra-icon css-1fnxnin"
            focusable="false"
            viewBox="0 0 16 16"
          >
            <path
              clip-rule="evenodd"
              d="M12.1632 10.2416C11.3318 10.2416 10.5918 10.6192 10.0755 11.2069L6.14368 8.88108C6.23509 8.59703 6.2969 8.29963 6.2969 7.98442C6.2969 7.67277 6.23596 7.37715 6.14541 7.09578L10.0711 4.78692C10.5874 5.37815 11.3292 5.75748 12.1632 5.75748C13.7156 5.75748 14.9789 4.46636 14.9789 2.87874C14.9789 1.29111 13.7156 0 12.1632 0C10.6109 0 9.34848 1.29111 9.34848 2.87874C9.34848 3.19038 9.40943 3.48511 9.49997 3.76738L5.57341 6.07624C5.05799 5.485 4.31534 5.10568 3.48214 5.10568C1.92892 5.10568 0.666504 6.39679 0.666504 7.98442C0.666504 9.57293 1.92892 10.8649 3.48214 10.8649C4.3136 10.8649 5.05364 10.4874 5.56906 9.89883L9.50171 12.2246C9.4103 12.5087 9.34848 12.8061 9.34848 13.1204C9.34848 14.708 10.6109 16 12.1632 16C13.7156 16 14.9789 14.708 14.9789 13.1204C14.9789 11.5327 13.7156 10.2416 12.1632 10.2416ZM12.1633 1.17535C13.0818 1.17535 13.8288 1.93933 13.8288 2.87873C13.8288 3.81812 13.0818 4.5821 12.1633 4.5821C11.2456 4.5821 10.4978 3.81812 10.4978 2.87873C10.4978 1.93933 11.2456 1.17535 12.1633 1.17535ZM3.48211 9.68868C2.56359 9.68868 1.81659 8.92469 1.81659 7.98441C1.81659 7.0459 2.56359 6.28103 3.48211 6.28103C4.39977 6.28103 5.14764 7.0459 5.14764 7.98441C5.14764 8.92469 4.39977 9.68868 3.48211 9.68868ZM12.1633 14.8247C11.2456 14.8247 10.4978 14.0608 10.4978 13.1205C10.4978 12.1811 11.2456 11.4171 12.1633 11.4171C13.0818 11.4171 13.8288 12.1811 13.8288 13.1205C13.8288 14.0608 13.0818 14.8247 12.1633 14.8247Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </a>
        <a
          href="#"
        >
          <svg
            class="chakra-icon css-1fnxnin"
            focusable="false"
            viewBox="0 0 16 16"
          >
            <path
              clip-rule="evenodd"
              d="M1.86885 15.9996C1.78287 15.9996 1.6977 15.9774 1.6182 15.932C1.443 15.8305 1.3335 15.6301 1.3335 15.412V1.44245C1.3335 0.646436 1.92319 0 2.64834 0H13.3528C14.0779 0 14.6668 0.646436 14.6668 1.44245V15.412C14.6668 15.6301 14.5565 15.8305 14.3813 15.932C14.2053 16.0335 13.9928 16.021 13.8297 15.8972L8.00016 11.5254L2.17058 15.8972C2.08055 15.9649 1.9751 15.9996 1.86885 15.9996ZM2.6483 1.17529C2.51365 1.17529 2.40415 1.2955 2.40415 1.44241V14.2989L7.69839 10.3277C7.88089 10.1906 8.12018 10.1906 8.30187 10.3277L13.5961 14.2989V1.44241C13.5961 1.2955 13.4866 1.17529 13.3528 1.17529H2.6483Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
